### PR TITLE
Adds RFC template based on the Rust RFC template

### PR DESCRIPTION
**Pull request by RCHowell**
*Issue #, if available:*

Related to https://github.com/partiql/partiql-docs/issues/5

*Description of changes:*

This PR adapts the Rust RFC template for use with the PartiQL RFC process

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by RCHowell** [comment]
I see this issue has not been merged. We could consider changing this to @dlurton's template
https://github.com/partiql/partiql-spec/pull/20/files#diff-78c4e145ff4726d177b2db60908abe387488d7343a454f6b42abe5d6dc5ad0f6

File: templates/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+
+# Summary
+[summary]: #summary
+
+One paragraph explanation of the feature.
```
Comment (am357) [inline templates/0000-template.md:7]: `... of the feature or the proposed change to the specification`?

File: templates/0000-template.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+the reader has proficient knowledge on the existing PartiQL specification.
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how PartiQL users should *think* about the feature, and how it should impact the way they use PartiQL. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing PartiQL users and new PartiQL users.
+
+For implementation-oriented RFCs, this section should focus on how PartiQL contributors should think about the change, and give examples of its concrete impact.
+For api-oriented RFCs, this section should focus on how PartiQL users would leverage or be affect by the changes.
```
Comment (am357) [inline templates/0000-template.md:27]: Suggestion: 

```
For spec-oriented RFCs, this section should focus on how implementations, based on the proposed spec. change will get impacted. It also should include any required grammar that accompanies with the spec. change. 
```

File: templates/0000-template.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Why is this design the best in the space of possible designs?
```
Comment (am357) [inline templates/0000-template.md:48]: maybe `design/proposal`?

File: templates/0000-template.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+[prior-art]: #prior-art
+
+Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- For specification proposals: Does this feature exist in SQL-92 or other SQL dialects?
```
Comment (am357) [inline templates/0000-template.md:58]: Suggestion: `SQL-92 -> any ISO SQL standard` b/c each ISO SQL standard replaces the previous one.

File: templates/0000-template.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how PartiQL users should *think* about the feature, and how it should impact the way they use PartiQL. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing PartiQL users and new PartiQL users.
+
+For implementation-oriented RFCs, this section should focus on how PartiQL contributors should think about the change, and give examples of its concrete impact.
+For api-oriented RFCs, this section should focus on how PartiQL users would leverage or be affect by the changes.
+
+# Reference-level explanation
```
Comment (am357) [inline templates/0000-template.md:29]: Thinking, this section might not be required for specification changes. If so, we might want to call that out.

**Review by am357 — COMMENTED** [review]
In addition to my comments inline: we might need change `design` to `design/proposal` or something similar in some of the sections to leave the scope open for spec. changes.

**Review by am357 — APPROVED** [review]

**Pull request merged**